bluetooth: RPC: GATT central over RPC implementation (#6134)
nrfconnect/sdk-nrf: doc/nrf/libraries/bluetooth_services/rpc.rst

   :class: highlight

   west build -b *board* -- -DCONFIG_OVERLAY=my_overlay_file.conf

API documentation
*****************

This library does not define a new API.
Instead, it uses Zephyr's Bluetooth API.

> The |NCS| currently supports serialization of:

Reviewer: ```suggestion
The |NCS| currently supports serialization of the following:
```
Personally, I would rather avoid the phrase "the following", but it is the company policy to have that.

   :class: highlight

   west build -b *board* -- -DCONFIG_OVERLAY=my_overlay_file.conf

> API documentation

Author: I added new APIs to the note. I think that it's safer to list APIs that we support (the list is not long).

As peknis suggested, we can have a link to the Zephyr's API in this section, so I will move list from note to this section, because we don't need duplication.

*****************

This library does not define a new API.
Instead, it uses Zephyr's Bluetooth API.
The behavior of the implementation is almost the same as Zephyr's with the following exceptions:

* The latency is longer because of the overhead for exchanging messages between cores.
* The :c:func:`bt_gatt_cancel` function is not implemented.
* The ``flags`` field of  the :c:struct:`bt_gatt_subscribe_params` structure is atomic, so it cannot be correctly handled by the nRF RPC.
  The library implements the following workaround for it:

>   * All ``flags`` are sent to the network core when either the :c:func:`bt_gatt_subscribe` or :c:func:`bt_gatt_resubscribe` function is called.
>      This covers most of the cases, because the ``flags`` are normally set once before those functions calls.

Reviewer: ```suggestion
  * All ``flags`` are sent to the network core when either the :c:func:`bt_gatt_subscribe` or :c:func:`bt_gatt_resubscribe` function is called.
    This covers most of the cases, because the ``flags`` are normally set once before those functions calls.
```

Reviewer: This one extra space made the doc render incorrectly, sorry my bad.

   :class: highlight

   west build -b *board* -- -DCONFIG_OVERLAY=my_overlay_file.conf

API documentation
*****************

This library does not define a new API.
Instead, it uses Zephyr's bluetooth API.
Behavior of the implementation is almost the same as Zephyr's with some exceptions listed below.

>    * The latency is bigger caused by overhead for exchanging messages between cores.
> 
>    * :c:func:`bt_gatt_cancel` is not implemented.
> 
>    * ``flags`` field of :c:struct:`bt_gatt_subscribe_params` is atomic, so it cannot be correctly handled by the nRF RPC.
>      The library implements following workaround for it:
> 
>       * All ``flags`` are send to the network core when :c:func:`bt_gatt_subscribe` or :c:func:`bt_gatt_resubscribe` are called.
>         This covers most of the cases, because ``flags`` normally are set once before those functions calls.
> 
>       * If you want to read or write ``flags`` after subscription you have to call
>         :c:func:`bt_rpc_gatt_subscribe_flag_set`, :c:func:`bt_rpc_gatt_subscribe_flag_clear` or :c:func:`bt_rpc_gatt_subscribe_flag_get`.

Reviewer: ```suggestion
* The latency is longer because of the overhead for exchanging messages between cores.
* The :c:func:`bt_gatt_cancel` function is not implemented.
* The ``flags`` field of  the :c:struct:`bt_gatt_subscribe_params` structure is atomic, so it cannot be correctly handled by the nRF RPC.
  The library implements the following workaround for it:

  * All ``flags`` are sent to the network core when either the :c:func:`bt_gatt_subscribe` or :c:func:`bt_gatt_resubscribe` function is called.
     This covers most of the cases, because the ``flags`` are normally set once before those functions calls.
  * If you want to read or write the ``flags`` after the subscription, you have to call :c:func:`bt_rpc_gatt_subscribe_flag_set`, :c:func:`bt_rpc_gatt_subscribe_flag_clear` or :c:func:`bt_rpc_gatt_subscribe_flag_get`.
```
